src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -308,7 +311,9 @@ public boolean validCheckForObject(final AtlasObject object)
                 recordedOsmIds.add(crossingItem.getOsmIdentifier());
             }
         }
-
+        final Set<Location> intersectingLocations = this.collectIntersectingLocations(atlas, object,
+                waterbody);
+        intersectingLocations.forEach(newFlag::addPoint);
```

Comment:
I would put this into the conditional above so that points are only added once for edges. Currently I think that each edge (two for each road) will add the same points to the flag. So just move this logic up into the conditional where the instructions are being added

Reply:
Addressed and pushed!